[#12125] Instructor results page stats tables: differentiate students with same team and name

ellaella12 opened the pull request:
Fixes #12125 
My attempt at adding email to statistics table. Not sure if I am supposed to modify the statistics-calculation files and subsequently the response.json test fiiles.
![image](https://user-images.githubusercontent.com/120079323/230439106-8f4fc411-d524-4e76-90d9-70647f115c2c.png)
![image](https://user-images.githubusercontent.com/120079323/230439203-e6db8663-c23a-462b-969e-5cdb4f40c764.png)
![image](https://user-images.githubusercontent.com/120079323/230439356-4184e08d-d9ae-4d9a-bb66-f269b4239c5f.png)
![image](https://user-images.githubusercontent.com/120079323/230440521-b033547b-20c6-428b-8b95-7d108ca29f61.png)


The emails was already included in the csv files before this feature
![image](https://user-images.githubusercontent.com/120079323/230441351-c2234573-f6a3-4aac-b660-d40a8a068411.png)

zhaojj2209 commented:
Do fix the failing E2E tests and update your branch with the changes from the master branch. Approach looks alright for now, we will do a more thorough review once the branch has been updated.

In the meantime, it would be good if you can add screenshots showing your fix, so as to help speed up the review process. Do also check if the same changes are present in the downloaded CSV data as well.

ellaella12 commented:
Hi I have updated the pr to pass the e2e tests, merge main into this branch, and added screenshots in the pull request. May I request for some help reviewing the pr? Thank you!

cedricongjh reviewed (commented):
Just a minor nit from my end!

ellaella12 commented on src/web/app/components/question-responses/per-question-view-responses/per-question-view-responses.component.html:
```diff
@@ -51,13 +51,15 @@
         <td *ngIf="showRecipient">
-          {{ response.recipient === '-' ? 'No Specific User' : response.recipient }}
+          {{ response.recipient === '-' ? 'No Specific User' : response.recipient}}
+          {{response.recipientEmail? " (" + response.recipientEmail + ")" : ""}}
```
not sure if I am correct, angular template doesn't seem to allow template literals. 
https://stackoverflow.com/questions/47057696/how-to-use-es6-template-literal-as-angular-component-input
i couldn't use {{...}} string interpolation again as it is used in the outer conditional expression

weiquu replied:
Yup I think you're right, ES6 template literals and nested string interpolation aren't supported. Could probably 1) move it out into a function then call it or 2) create a separate element for it with:
`<ng-container *ngIf="response.recipientEmail">({{ response.recipientEmail }})</ng-container>`

Leaning towards the latter, but tbh I don't think we need to go so far to avoid string concatenation. What do you all think?

samuelfangjw replied:
The simple solution will suffice here, shouldn't overcomplicate things.

weiquu replied:
Okay! Resolving this convo then

weiquu reviewed (approved):
LGTM

cedricongjh reviewed (approved):
LGTM! thank you for your contribution to TEAMMATES
